src/app/components/BCP47Tag.svelte

```diff
@@ -3,22 +3,27 @@
   import * as bcp47 from "bcp-47";
   import worker from "../spawn-worker";
 
-  const TAG_INPUT_ID = "tag-input";
+  let errorMessage: string = "";
 
-  $: tag = "";
-  $: schema = bcp47.parse(tag);
-  $: validTag = tag === "" || (tag.length > 0 && schema.language !== null);
-  $: error = validTag ? "" : "Invalid BCP 47 Tag";
-  $: if (validTag) {
-    worker.setProjectData({ languages: [{ name: "", id: tag }] });
+  function onInputValue(event: CustomEvent) {
```

Comment:
Could be called something like... this?

```suggestion
  function validateAndSetLanguage(event: CustomEvent) {
```